specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -1960,6 +2312,119 @@
         "id"
       ],
       "description": "Virtual Network rule."
+    },
+    "Snapshot": {
+      "type": "object",
+      "description": "Response for Volume Snapshot request.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of Volume Snapshot.",
+          "$ref": "#/definitions/SnapshotProperties"
+        },
+        "systemData": {
+          "description": "Resource metadata required by ARM RPC",
+          "readOnly": true,
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```

Comment:
>"#/definitions/Resource" [](http://example.com/codeflow?start=18&length=24)

Get this from v5 common types. Same comment for SnapshotCreateParameter.

Reply:
And usually its preferable to inherit specifically ProxyResource or TrackedResource than Resource.